hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheCreateConfigMessageTask.java

```diff
@@ -56,7 +76,7 @@ protected Operation prepareOperation() {
     protected ClientMessage encodeResponse(Object response) {
         CacheService service = getService(getServiceName());
         final Data responseData = service.toData(response);
-        return CacheCreateConfigCodec.encodeResponse(serializationService.toData(responseData));
+        return CacheCreateConfigCodec.encodeResponse(responseData);
```

Comment:
Client 3.6.5 with member 3.8-SNAPSHOT breaks for me here; we are missing the compatibility flag check to respond with the appropriate `CacheConfig` type.